lib/services/local/access_list.go

```diff
@@ -189,7 +189,12 @@ func (a *AccessListService) UpsertAccessList(ctx context.Context, accessList *ac
 				ownerMap[owner.Name] = struct{}{}
 			}
 
-			upserted, err = a.service.UpsertResource(ctx, accessList)
+			if oldAccessList == nil {
+				upserted, err = a.service.UpsertResource(ctx, accessList)
+				return trace.Wrap(err)
+			}
+
+			upserted, err = a.service.ConditionallyUpdateResource(ctx, accessList)
```

Comment:
I'm not sure we can get around this. In order to not have a potential security leak, we *must* always lock. Otherwise there's a (extremely small) potential delta where a user who shouldn't have access to an access list is able to modify it.